src/webgpu/shader/execution/builtin/cos.spec.ts

```diff
@@ -0,0 +1,40 @@
+export const description = `
+Execution Tests for the 'cos' builtin function
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { F32, TypeF32 } from '../../../util/conversion.js';
+
+import { absThreshold, Case, Config, run } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('float_builtin_functions,cos')
+  .uniqueId('650973d690dcd841')
+  .specURL('https://www.w3.org/TR/2021/WD-WGSL-20210929/#float-builtin-functions')
+  .desc(
+    `
+cos:
+T is f32 or vecN<f32> cos(e: T ) -> T Returns the cosine of e. Component-wise when T is a vector. (GLSLstd450Cos)
+
+Please read the following guidelines before contributing:
+https://github.com/gpuweb/cts/blob/main/docs/plan_autogen.md
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const cases = new Array<Case>(1000);
+    for (let i = 0; i < cases.length; i++) {
+      const angle = -Math.PI + (2.0 * Math.PI * i) / cases.length;
```

Comment:
This is the easy range for cosine.  Yes, it's periodic, but sometimes the hard part is in doing the range reduction accurately.  Values nearest the largest positive and negative finite values would be interesting too.

Suggestions:
- divide by cases.length-1  so the +Math.PI case is tested too


Please reference https://github.com/gpuweb/cts/issues/792 so this sampling can be visited.  With a reference like that, I'm ok landing this for now.

Reply:
The reference should be listed as TODO so it can be picked up automatically later.

Reply:
Done.